src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java

```diff
@@ -185,34 +163,31 @@ public double getPermanentLimit2() {
         return branch.getCurrentLimits2() != null ? branch.getCurrentLimits2().getPermanentLimit() * getBus2().getNominalV() / PerUnit.SB : Double.NaN;
     }
 
-    @Override
-    public Optional<PhaseControl> getPhaseControl() {
-        return Optional.ofNullable(phaseControl);
-    }
-
     @Override
     public void updateState() {
         branch.getTerminal1().setP(p1.eval() * PerUnit.SB);
         branch.getTerminal1().setQ(q1.eval() * PerUnit.SB);
         branch.getTerminal2().setP(p2.eval() * PerUnit.SB);
         branch.getTerminal2().setQ(q2.eval() * PerUnit.SB);
 
-        if (phaseControl != null) { // it means there is a regulating phase tap changer
+        if (isPhaseController()) { // it means there is a regulating phase tap changer
```

Comment:
@floriand-e2r Do not forget to add condition && phaseControl.getMode() == PhaseControl.Mode.OFF